src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,145 @@
+        FileWriter fw = new FileWriter(result);
+        BufferedWriter bw = new BufferedWriter(fw);
+
+        // Lectura de datos
+        String infile = path;
+        GZIPInputStream in = new GZIPInputStream(new FileInputStream(infile));
+        Reader decoder = new InputStreamReader(in);
+        BufferedReader br = new BufferedReader(decoder);
+        String line;
+
+        String cvsProd="";
+        String cvsUser="";
+        String cvsScore = "";
+
+        int prodNum = 0;
+        int userNum = 0;
+        int reviewNum = 0;
+
+
+        while ((line = br.readLine()) != null) {
+            if (line.startsWith("product/productId:")) {
+                reviewNum++;
+
+                String [] parts = line.split(": ");
+                String productId = parts[1];
+                
+                if (!productsMap.containsKey(productId)) {
+                    productsMap.put(productId,prodNum++);
+                }
+                cvsProd = productsMap.get(productId).toString();
+
+            } else if (line.startsWith("review/userId")) {
+                String [] parts = line.split(": ");
+                String userId = parts[1];
+
+                if(!usersMap.containsKey(userId)) {
+                    usersMap.put(userId,userNum++);
+                }
+                cvsUser = usersMap.get(userId).toString();
+
+            } 
+            else if (line.startsWith("review/score")) {
+                String [] parts = line.split(": ");
+                cvsScore = parts[1];
+
+                String entry = String.format("%s,%s,%s\n",   cvsUser ,cvsProd, cvsScore);
+                bw.write(entry);
+            }
+        }
+        
+        br.close();
+        bw.close();
+
+        this.totalReviews = reviewNum;
+        this.totalProducts = productsMap.size();
+        this.totalUsers = usersMap.size();
+
+    }
+
+    public int getTotalReviews() {
+        return this.totalReviews;
+    }
+
+    public int getTotalProducts() {
+        return this.totalProducts;
+    }
+
+    public int getTotalUsers() {
+        return this.totalUsers;
+    }
+
+    
+    public List<String> getRecommendationsForUser(String userId) throws IOException, TasteException {
+
+        List<String> results = new ArrayList<String>();
+
+        int user = usersMap.get(userId);
+
+        DataModel model = new FileDataModel(new File("Result.csv"));
+        UserSimilarity similarity = new PearsonCorrelationSimilarity(model);
+        UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+        UserBasedRecommender recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+        List<RecommendedItem> recommendations = recommender.recommend(user, 3);
+
+        for (RecommendedItem recommendation : recommendations) {
+            results.add(productsMap.inverse().get((int)recommendation.getItemID()));
+        }
+        return results;
+
+    }
```

Comment:
This function is another example of the class handling too many responsibilities.